db/silkworm/stagedsync/stagedsync.hpp

```diff
@@ -0,0 +1,85 @@
+/*
+   Copyright 2020 - 2021 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+           http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+#include <silkworm/db/tables.hpp>
+#include <silkworm/db/chaindb.hpp>
+#include <silkworm/stagedsync/util.hpp>
+#include <vector>
+
+#ifndef SILKWORM_STAGEDSYNC_HPP_
+#define SILKWORM_STAGEDSYNC_HPP_
+
+namespace silkworm::stagedsync {
+
+extern bool write_receipts;
+constexpr size_t kBatchSize = 512 * kMebi;
```

Comment:
`write_receipts` & `kBatchSize` shouldn't be in `db/silkworm/stagedsync/stagedsync.hpp` since it's not something pertaining to all stages, but specific for a particular stage.

Reply:
write_receipts needs to be modified in case receipts are disabled or not outside of stage_execution, and that is why it is an extern. kBatchSize is both used in stage_senders and stage_execution.

Reply:
Fine about `kBatchSize`, we can have it in `stagedsync.hpp` for now.

But `write_receipts` shouldn't be there. The current logic in master in `cmd/execute.cpp` is
```
bool write_receipts{db::read_storage_mode_receipts(*txn)};
```
Let's keep the current logic and obtain the value of `write_receipts` from the database in the execution code (no need for an `extern` variable).

Reply:
oh, i did not notice that